src/mail_room.py

```diff
@@ -0,0 +1,166 @@
+#  _*_ coding:utf-8 _*_
+"""A simple python program to keep track of names and donations."""
+from __future__ import print_function
+import sys
+import io
+
+
+GIFTER_DICT = {}
+
+prompts = ["Would you like to 'Send a Thank you' or 'Create a Report'?", "Enter the donator name or press 'L' to see a list of previous names", "How much is the donation?"]
+email_text = "Dear {},\nThank you so much for you donation of ${}. You are a good person.\n"
+
+
+def main_controller():
+    """Control flow of program."""
+    begin = send_or_report()
+    if begin is True:
+        name = get_name()
+        check_dict(name, GIFTER_DICT)
+        ammt = get_ammt()
+        update_dict(name, ammt)
+        send_email(name, ammt)
+        main_controller()
```

Comment:
Calling this same function recursively is not a good choice for making a piece of code repeat.  The problem is that every time you re-execute the function, you add a new frame to the execution stack.  

I added a line of code to this function at the beginning to print out the number of frames in the stack each time this function is called.  Here's what my interaction with your program printed out:

```
The stack is 2 frames deep
Would you like to 'Send a Thank you' or 'Create a Report'?
>>>>>> Create a Report
there is nothing to report
The stack is 3 frames deep
Would you like to 'Send a Thank you' or 'Create a Report'?
>>>>>> Send a Thank you
Enter the donator name or press 'L' to see a list of previous names
>>>>>> Cris Ewing
How much is the donation?
>>>>>> 25
Dear cris ewing,
Thank you so much for you donation of $25. You are a good person.

The stack is 4 frames deep
Would you like to 'Send a Thank you' or 'Create a Report'?
>>>>>> Send a thank you
Enter the donator name or press 'L' to see a list of previous names
>>>>>> l
['cris ewing']
Enter the donator name or press 'L' to see a list of previous names
>>>>>> cris ewing
How much is the donation?
>>>>>> 24
Dear cris ewing,
Thank you so much for you donation of $24. You are a good person.

The stack is 5 frames deep
Would you like to 'Send a Thank you' or 'Create a Report'?
>>>>>> Create a Report
Total:$49 Name: cris ewing Number of Times: 2 Average:$24.5
```

See how the count of frames increases each  time I come back to this function?  That will never end.  Your program has no way to "unwind" the stack and return to the original calling context.  

Can you think of a different way to repeat the steps in this code without using recursion?